### PR TITLE
fix: allow search filter to use an int for its value

### PR DESCRIPTION
**Pull request by jmontoyaa**
| Q             | A
| ------------- | ---
| Branch       | 2.6
| Tickets       | #1633  #4196 #2190 #2308
| License       | MIT

Same fix as #4290 but with tests and  in branch 2.6

### CODE REVIEW

File: src/Bridge/Doctrine/Common/Filter/SearchFilterTrait.php [hunk 1]
```diff
@@ -136,7 +136,7 @@ protected function getIdFromValue(string $value)
     protected function normalizeValues(array $values, string $property): ?array
     {
         foreach ($values as $key => $value) {
-            if (!\is_int($key) || !\is_string($value)) {
+            if (!\is_int($key) || \is_array($value)) {
```
Comment (antograssiot) [inline src/Bridge/Doctrine/Common/Filter/SearchFilterTrait.php:139]: What value will not fit the first part of the IF then ? I mean an array will already be validated by the first part of the OR condition isn't it ?

Comment (jmontoyaa) [reply src/Bridge/Doctrine/Common/Filter/SearchFilterTrait.php:139]: First part of the IF it's the $key, second part it's about the $value. If $value is an INT then a notice will appear. I wrote a test about that here: 

https://github.com/api-platform/core/pull/4295/commits/d33159fc1af1c9ac3165a811ffa30c8f8d7cdf7c

File: features/graphql/collection.feature [hunk 2]
```diff
@@ -909,3 +909,65 @@ Feature: GraphQL collection support
     Then the response status code should be 200
     And the response should be in JSON
     And the JSON node "data.fooDummies.collection" should have 1 element
+
+  @createSchema
+  Scenario: Retrieve a collection with a nested collection filtering by age through a GraphQL query
```
Comment (alanpoulain) [inline features/graphql/collection.feature:914]: Wouldn't it be better in `filters.feature` instead?

File: features/graphql/collection.feature [hunk 3]
```diff
@@ -909,3 +909,65 @@ Feature: GraphQL collection support
...
+                  age
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+
```
Comment (alanpoulain) [inline features/graphql/collection.feature:939]: Line to remove.

File: features/graphql/collection.feature [hunk 4]
```diff
@@ -909,3 +909,65 @@ Feature: GraphQL collection support
...
+    Then the JSON node "data.dummies.totalCount" should be equal to 1
+    And the JSON node "data.dummies.edges[0].node.relatedDummies.totalCount" should be equal to 1
+    Then the JSON node "data.dummies.edges[0].node.relatedDummies.edges[0].node.age" should be equal to "31"
```
Comment (alanpoulain) [inline features/graphql/collection.feature:942]:
```suggestion
    And the JSON node "data.dummies.edges[0].node.relatedDummies.edges[0].node.age" should be equal to "31"
```

**Comment by alanpoulain** [comment]
Thank you for the Behat tests! I think it would be nice to have unit ones too. I will handle them.

**Comment by jmontoyaa** [comment]
> Thank you for the Behat tests! I think it would be nice to have unit ones too. I will handle them.

Thanks! I just added fixed your suggestions.

**Comment by jmontoyaa** [comment]
I just fix for the behat test (subresource.feature) that was failing because the age filter was added.

**Review by alanpoulain — APPROVED** [review]

**Comment by alanpoulain** [comment]
Thank you @jmontoyaa.

**Pull request merged**